frontend/src/ui/Blocks.tsx

```diff
@@ -57,14 +59,60 @@ type TextProps = {
     content: string;
 };
 
+// We don't just want to allow everything, so here we explicitly list what we
+// support. Notably missing:
+//
+// - img (we might want to allow them but need to figure out storage first)
+// - headings (if we allow them, we need to map `#` to h2 or h3 so that they
+//   don't interfere with other headings on the page.)
+const ALLOWED_MARKDOWN_TAGS = [
+    "p", "blockquote", "pre", "ul", "ol", "li", "a", "em", "strong", "code", "hr",
+];
+
+const CODE_BACKGROUND_COLOR = "#f9f9f6";
+
+// We override some components emitted by the Markdown parser to add CSS.
+const MARKDOWN_COMPONENTS: Components = {
+    p: ({ node: _, ...props }) => <p css={{ margin: "16px 0" }} {...props} />,
+    blockquote: ({ node, ...props }) => <blockquote
+        css={{
+            borderLeft: "4px solid #e5e5e5",
+            padding: "2px 8px",
+            "& > *:first-child": { marginTop: 0 },
+            "& > *:last-child": { marginBottom: 0 },
+        }}
+        {...props}
+    />,
+    code: ({ node, className, inline, ...props }) => <code
+        css={inline === true && {
```

Comment:
is `inline: boolean`? or `boolean?`? In any case, `inline === true` should be equivalent to `inline`:

```suggestion
        css={inline && {
```

Reply:
Similar to above, with this I get:

```
error  Unexpected nullable boolean value in conditional. Please handle the nullish case explicitly  @typescript-eslint/strict-boolean-expressions
```

Reply:
Oof, that sucks even more. But it makes sense. We could apparently [configure the rule](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/strict-boolean-expressions.md) to ignore certain types, but I honestly wouldn't, I think. :thinking: